### ISSUE TITLE
need clarification in documentation

### ISSUE THREAD

**Issue by leo-gan** [Microsoft/LightGBM#356]
in https://github.com/Microsoft/LightGBM/blob/master/docs/Parameters.md

- "fair, Fair loss" - link here is broken and didn't point to the explanation;
- "fair_c, default=1.0, type=double ... parameter for Fair loss." - as previous
- "lambdarank, lambdarank application" - there is no link to the explanation. Google didn't bring anything meaningful;
- "goss, Gradient-based One-Side Sampling" - there is no link to the explanation. Google didn't bring anything meaningful;

also in Metric Parameters:
- "fair, Fair loss" - as previous;
- "auc, AUC" - link is wrong, it should be [https://en.wikipedia.org/wiki/Receiver_operating_characteristic#Area_under_the_curve]
- "binary_logloss, log loss" - link to the empty page

**Comment by wxchan** [comment]
fair: @henry0312 
lambdarank: https://pdfs.semanticscholar.org/fc9a/e09f9ced555558fdf1e997c0a5411fb51f15.pdf
goss: I think it's not published yet? @guolinke

**Comment by Laurae2** [comment]
For simple understanding:

* Fair loss: derivable proxy for MAE objective function, check https://www.kaggle.com/c/allstate-claims-severity/discussion/24520 for more details about MAE proxies
* Fair loss c parameter: check https://www.kaggle.com/c/allstate-claims-severity/discussion/24520 - bigger c induces more brutal transitions (closer to V curve), while smaller c induces a smoother transition (closer to U curve) between the expected proxy MAE and the real MAE (U curve vs V curve)
* LambdaRank: objective function for ranking tasks (its superset algorithm is LambdaMART - you might find more information about it on Google)
* Gradient-based One-Side Sampling: an alternative boosting method which seems to be performing Gradient sampling (according to the [code](https://github.com/Microsoft/LightGBM/blob/master/src/boosting/goss.hpp)), thus it is an adaptive method exploiting information (N.B: as it is an adaptive method, you cannot use bagging)

**Comment by henry0312** [comment]
I'm sorry for my late response.
However, I can't find another documetns for fair loss 😢

**Comment by wxchan** [comment]
the loss function name is really unfriendly for search engine, and seem this loss function is not mentioned a lot too
